.gitignore

```diff
@@ -1,6 +1,7 @@
 *.egg
 *.egg-info
 *.pyc
+.idea
```

Comment:
please add these to a [global gitignore](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore) rather than the project

Reply:
Unfortunately, this belongs in the project gitignore and cannot go in the global gitignore because some projects require some of the files from this directory to be in Git.

See https://intellij-support.jetbrains.com/hc/en-us/articles/206544839 for more info.

Reply:
Eh? That you've put it in `.gitignore` explicitly states those files should _not_ be tracked in git. IntelliJ is a detail of your preferred development environment, and not Pyflakes. That's why it should go in your personal .gitignore, not Pyflakes.

I use Vim which puts a `.swp` file next to whatever I'm editing. Some people use editors which make backups with a `~` suffix. But those aren't in here either.

Reply:
Exactly, I'm saying none of those files from the .idea dir should be in *this* project's repo.

But some *other* projects do want to share IDE config via some files from the .idea dir. So it's not a global level thing, it's a project-level thing. 

I assume Pyflakes does not want any of that config, hence ignoring the entire dir in this case. 

What sort of problems does it cause to have an extra entry in this gitignore?

Reply:
@hugovk when people insist upon thrusting their personal editor's junk into project gitignore files, it invites lots of unwelcome and unhelpful pull requests to add ever more editor junk files to the gitignore, usually by people who otherwise don't contribute to a project. I'm in agreement with @bitglue that this doesn't belong in this project's `.gitignore`. If projects want to track that information, then it can be added inspite of your global gitignore. What your global gitignore prevents, however, is adding it blindly with `git add .`.

Reply:
You can also follow the instructions for a global gitignore above, but omit the `--global` option. This will change the setting just for the repo, so (just for pyflakes) you can use an alternative .gitignore. I'm not sure if that supersedes or arguments the default .gitignore, but you may want to experiment.

You might also be able to put a fully qualified path in your global .gitignore to ignore `.idea` in pyflakes, but not your other git repositories. Again I'm not sure this works, but something to try.

Prepending a leading `!` to a line in gitignore can be used to unignore a file. I'm not familiar with the exact semantics of this feature as I've not had reason to do use it, but perhaps you could add it to projects for which sharing IDE files is explicitly desired.

In any case, the `.gitignore` here does not currently contain any entries that reflect developer preference: only those due to pyflakes and its associated tooling in this repository. That's unlikely to change.

Regardless, thank you for the contributions. They are very much appreciated. I should be able to get this merged and released soon.

Reply:
Aha, thanks for the tips on unignoring the global ignore, I'll give that a shot! I've removed it from this project's one!